include/motor-control/core/stepper_motor/motor_interrupt_handler.hpp

```diff
@@ -221,9 +221,36 @@ class MotorInterruptHandler {
         set_buffered_move(Move{});
     }
 
-    auto get_encoder_pulses() { return hardware.get_encoder_pulses(); }
+    auto get_encoder_pulses() {
+        if (get_encoder_overflow_status() == true) {
+            hardware.clear_encoder_SR();
+            if (enc_overflow_future_flag) {
+                enc_position_tracker = hardware.get_encoder_pulses();
+                enc_overflow_future_flag = false;
+            } else {
+                enc_position_tracker =
+                    UINT16_MAX + hardware.get_encoder_pulses();
+                enc_overflow_future_flag = true;
+            }
+        } else {
+            enc_position_tracker = hardware.get_encoder_pulses();
+        }
+        return enc_position_tracker;
+        /* This function fixes the overflow issue for each motor.
+        Check whether the UIEF interrupt bit in the Status register gets
+        triggered. If it gets triggered we save the previous state of the
+        triggered flag and clear the status register interrupt bit.
+        */
+    }
 
     void reset_encoder_pulses() { hardware.reset_encoder_pulses(); }
+
+    void clear_encoder_overflow_flag() { hardware.clear_encoder_SR(); }
```

Comment:
these one-liner functions are a bit confusing, could we just call the hardware methods directly? Or, if you want to stick with functions like this, you should make sure you use them everywhere.

Reply:
We can stick with just hardware methods.